uaclient/cli.py

```diff
@@ -124,9 +127,13 @@ def assert_lock_file(lock_holder=None):
     def wrapper(f):
         @wraps(f)
         def new_f(*args, cfg, **kwargs):
-            with lock.SingleAttemptLock(cfg=cfg, lock_holder=lock_holder):
-                retval = f(*args, cfg=cfg, **kwargs)
-            return retval
+            try:
+                with lock.SingleAttemptLock(cfg=cfg, lock_holder=lock_holder):
+                    retval = f(*args, cfg=cfg, **kwargs)
+                return retval
+            except exceptions.LockHeldError as e:
+                event.raise_error_and_finish(exception=e)
+                return 1
```

Comment:
Can some of the places where this is inserted be removed if we instead call `event.raise_error` in `main_error_handler`?

Reply:
The problem about doing that on `main_error_handler` is that we would need to verify there as well if the `args.format` is a json type to print the json output to the user. But not all commands support that, so it will not gonna that generic

Reply:
We default to CLI mode though, and we set the global mode variable to json if the `@set_event_mode` decorator is called and sees the json format option.
By the time we get to `main_error_handler` then the mode will have been set to json or is still at default cli - and I think will be accurate, right? Because for commands that don't support `--format` then it will just default to cli mode.
That means the `print` does have to be _in_ the event module though (contrary to my opinion in a different comment).

Reply:
Yes, you are actually right, I was worried that the `main_error_handler` decorator is called first, but it doesn't matter since the state will still be as MACHINE_READABLE. Good suggestion, I will refactor that bit then